sched/semaphore/sem_holder.c

```diff
@@ -891,6 +789,8 @@ void nxsem_restore_baseprio(FAR struct tcb_s *stcb, FAR sem_t *sem)
               (sem->semcount <= 0 && stcb != NULL));
 #endif
 
+  DEBUGASSERT(!up_interrupt_context());
```

Comment:
why can't be called from the interrupt context?

Reply:
The same as `nxsem_checkholder` -> `if (up_interrupt_context()) { return false; }`